### PR TITLE
Add an command line flag to DafnyLanguageServer.exe to control caching

### PR DESCRIPTION
**Pull request by cpitclaudel**
I'd like to collect data on how well the caching feature works, and part of that would be making it easy to turn it on and off in the UI.  But to do that I need it to be exposed on the CLI first, hence this PR.

### CODE REVIEW

File: Source/DafnyLanguageServer/Language/DafnyProgramVerifier.cs [hunk 1]
```diff
@@ -45,16 +45,21 @@ public class DafnyProgramVerifier : IProgramVerifier {
           // https://github.com/boogie-org/boogie/blob/b03dd2e4d5170757006eef94cbb07739ba50dddb/Source/VCGeneration/Couterexample.cs#L217
           DafnyOptions.O.ModelViewFile = "-";
           DafnyOptions.O.VcsCores = GetConfiguredCoreCount(options.Value);
+          DafnyOptions.O.VerifySnapshots = (int)options.Value.VerifySnapshots;
```
Comment (MikaelMayer) [inline Source/DafnyLanguageServer/Language/DafnyProgramVerifier.cs:48]: Is this PR still a draft? If so, mark is as such. It seems that you are defining an option but never using it anywhere except in loggin.

**Review by MikaelMayer — CHANGES_REQUESTED** [review]

File: Source/DafnyLanguageServer/Language/DafnyProgramVerifier.cs (hunk 1 above)
Comment (cpitclaudel) [reply Source/DafnyLanguageServer/Language/DafnyProgramVerifier.cs:48]: It's not a draft, it's ready to go.

Comment (MikaelMayer) [reply Source/DafnyLanguageServer/Language/DafnyProgramVerifier.cs:48]: What does it enable that was not possible before?

Comment (cpitclaudel) [reply Source/DafnyLanguageServer/Language/DafnyProgramVerifier.cs:48]: The option is set through the command line, and the command line flag is defined automatically by reflection.

Comment (cpitclaudel) [reply Source/DafnyLanguageServer/Language/DafnyProgramVerifier.cs:48]: It lets you turn on caching in Dafny's LSP server

**Review by keyboardDrummer — APPROVED** [review]

**Review by keyboardDrummer — APPROVED** [review]

File: Source/DafnyLanguageServer/Language/DafnyProgramVerifier.cs (hunk 1 above)
Comment (MikaelMayer) [reply Source/DafnyLanguageServer/Language/DafnyProgramVerifier.cs:48]: Oh I see, so it's a Boogie option that needed to be declared by Dafny.

**Review by MikaelMayer — APPROVED** [review]

**Pull request merged**